lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -482,14 +516,146 @@ namespace Intl
         error = UErrorCode::U_ZERO_ERROR;
 
         written = uloc_toLanguageTag(defaultLocaleId, bcp47, ULOC_FULLNAME_CAPACITY, true, &error);
-        ICU_ASSERT_RETURN(error, written > 0 && written < cchLangtag, 0);
+        ICU_ASSERT_RETURN(error, written  > 0 && ((size_t) written) < cchLangtag, 0);
 
         return utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(
             langtag,
             reinterpret_cast<LPCUTF8>(bcp47),
             reinterpret_cast<LPCUTF8>(bcp47 + written)
         );
     }
+
+    bool CollatorGetCollation(_In_z_ const char *langtag, _Out_ char *collation, _In_ size_t cchCollation)
+    {
+        Assert(langtag != nullptr && collation != nullptr && cchCollation >= 0 && cchCollation < ULOC_FULLNAME_CAPACITY);
+        collation[0] = 0;
+
+        bool ret = false;
+
+        UErrorCode error = U_ZERO_ERROR;
+        char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int32_t length = 0;
+        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &error);
+        ICU_ASSERT_RETURN(error, length > 0, ret);
+
+        UCollator *collator = ucol_open(localeID, &error);
+        ICU_ASSERT_RETURN(error, true, ret);
+
+        const char *collatorLocaleID = ucol_getLocaleByType(collator, ULOC_VALID_LOCALE, &error);
+        ICU_ASSERT_GOTO(error, true, LCloseCollator);
+
+        char collatorCollation[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int32_t collatorCollationLength = uloc_getKeywordValue(collatorLocaleID, "collation", collatorCollation, _countof(collatorCollation), &error);
+        ICU_ASSERT_GOTO(error, collatorCollationLength >= 0, LCloseCollator);
+        if (collatorCollationLength == 0)
+        {
+            // if the collation value doesn't exist, that is fine
```

Comment:
> that is fine [](start = 52, length = 13)

Can you explain this a bit more in the comment? Fine in what sense? We "succeeded" to find out that it definitely doesn't exist? (i.e. false = failure but not-exist is not a failure?) At this point "collation" is an empty string, and we're saying that's okay?

Reply:
It means that the locale simply didn't have a -u-co on it. This is checked for in Intl.js, where the return of this function is called `maybeCollation`